src/com/amplitude/api/AmplitudeClient.java

```diff
@@ -126,85 +144,111 @@ public void run() {
+    public AmplitudeClient enableLocationListening() {
         if (deviceInfo == null) {
             throw new IllegalStateException(
                     "Must initialize before acting on location listening.");
         }
         deviceInfo.setLocationListening(true);
+        return instance;
     }
 
-    public void disableLocationListening() {
+    public AmplitudeClient disableLocationListening() {
         if (deviceInfo == null) {
             throw new IllegalStateException(
                     "Must initialize before acting on location listening.");
         }
         deviceInfo.setLocationListening(false);
+        return instance;
     }
 
-    public void setEventUploadThreshold(int eventUploadThreshold) {
+    public AmplitudeClient setEventUploadThreshold(int eventUploadThreshold) {
         this.eventUploadThreshold = eventUploadThreshold;
+        return instance;
     }
 
-    public void setEventUploadMaxBatchSize(int eventUploadMaxBatchSize) {
+    public AmplitudeClient setEventUploadMaxBatchSize(int eventUploadMaxBatchSize) {
         this.eventUploadMaxBatchSize = eventUploadMaxBatchSize;
         this.backoffUploadBatchSize = eventUploadMaxBatchSize;
+        return instance;
     }
 
-    public void setEventMaxCount(int eventMaxCount) {
+    public AmplitudeClient setEventMaxCount(int eventMaxCount) {
         this.eventMaxCount = eventMaxCount;
+        return instance;
     }
 
-    public void setEventUploadPeriodMillis(int eventUploadPeriodMillis) {
+    public AmplitudeClient setEventUploadPeriodMillis(int eventUploadPeriodMillis) {
         this.eventUploadPeriodMillis = eventUploadPeriodMillis;
+        return instance;
     }
 
-    public void setMinTimeBetweenSessionsMillis(int minTimeBetweenSessionsMillis) {
+    public AmplitudeClient setMinTimeBetweenSessionsMillis(long minTimeBetweenSessionsMillis) {
         this.minTimeBetweenSessionsMillis = minTimeBetweenSessionsMillis;
+        return instance;
     }
 
-    public void setSessionTimeoutMillis(long sessionTimeoutMillis) {
+    public AmplitudeClient setSessionTimeoutMillis(long sessionTimeoutMillis) {
         this.sessionTimeoutMillis = sessionTimeoutMillis;
+        return instance;
     }
 
-    public void setOptOut(boolean optOut) {
+    public AmplitudeClient setOptOut(boolean optOut) {
         this.optOut = optOut;
 
         SharedPreferences preferences = context.getSharedPreferences(
                 getSharedPreferencesName(), Context.MODE_PRIVATE);
         preferences.edit().putBoolean(Constants.PREFKEY_OPT_OUT, optOut).commit();
+        return instance;
     }
 
-    public void setOffline(boolean offline) {
+    public AmplitudeClient setOffline(boolean offline) {
         this.offline = offline;
 
         // Try to update to the server once offline mode is disabled.
         if (!offline) {
             uploadEvents();
         }
+
+        return instance;
+    }
+
+    public AmplitudeClient trackSessionEvents() {
```

Comment:
Originally it had a boolean input but I opted to take it out to keep it consistent with some of our other public methods like useAdvertisingIdForDeviceId, enableLocationListening, which toggles internal booleans to true. Although I just saw enableNewDeviceIdPerInstall which does take in a boolean input. I dunno, what do you think?